lib/collection/property-list.js

```diff
@@ -266,7 +272,7 @@ _.extend(PropertyList.prototype, /** @lends PropertyList.prototype */ {
      */
     map: function (iterator, context) {
         !context && (context = this);
-        return _.map(this.members, iterator, context);
+        return _.map(this.members, iterator.bind(context));
```

Comment:
@kunagpal Did you read this comment correctly?

@shamasis Has mentioned that you need to do:
```
context ? iterator.bind(context) : iterator
```

What you have done is:
```
_.isFunction(iterator) ? iterator.bind(context || this) : iterator);
```

That is entirely different.
/cc @shamasis